pandas/io/common.py

```diff
@@ -583,12 +638,15 @@ def __init__(
         self.archive_name = archive_name
         kwargs_zip: Dict[str, Any] = {"compression": zipfile.ZIP_DEFLATED}
         kwargs_zip.update(kwargs)
-        super().__init__(file, mode, **kwargs_zip)
+        super().__init__(file, mode, **kwargs_zip)  # type: ignore[arg-type]
```

Comment:
complains about `file` being `IOBase` but we cannot have an `assert not isisntance(file, IOBase)` since `io.StringIO` inherits from `IOBase`